File: cpp/demos/demo_kinodynamic_rrt_star.cpp
#include "demo_common.hpp"
#include "navigation/global_planning/sampling/kinodynamic_rrt_star.hpp"

int main(int argc, char** argv) {
  try {
    demo::Args args = demo::parse_args(argc, argv);
    auto params = navigation::core::ParamSet::from_yaml(args.params);
    navigation::global_planning::KinodynamicRrtStarPlanner planner(params);
    // Binds the grid as a SamplingSpace<Point>&; the planner owns its double-integrator
    // dynamics and queries only (x, y) validity, so the sampling demo flow applies.
    return demo::run_sampling(args, params, planner);
  } catch (const std::exception& e) {
    std::cerr << "error: " << e.what() << "\n";
    return 1;
  }
}

File: python/demos/demo_kinodynamic_rrt_star.py
#!/usr/bin/env python3
"""Kinodynamic RRT* demo (Webb & van den Berg 2013).

Binds the occupancy grid as a `SamplingSpace` (the planner owns its double-integrator
dynamics and only queries (x, y) validity), so it reuses the sampling demo flow.
"""

from __future__ import annotations

from demo_common import run_sampling

# Imported by full module path: the sampling package __init__ is not edited by this
# feature, so the class is not re-exported from navigation.global_planning.
from navigation.global_planning.sampling.kinodynamic_rrt_star import KinodynamicRRTStar

if __name__ == "__main__":
    run_sampling("kinodynamic_rrt_star", KinodynamicRRTStar)
